compiler/src/iree/compiler/Dialect/TensorExt/IR/TensorExtOps.cpp

```diff
@@ -554,6 +561,239 @@ OpFoldResult CastToRaggedShapeOp::getNumRaggedRowsAsOfr() {
+      getValueOrCreateConstantIndexOp(rewriter, loc, givenRange[0].stride);
+
+  // Replace the `memref.dim`/`tensor.dim` operation in the backward slice of
+  // the upper bound with the number of ragged rows.
+  FailureOr<SmallVector<Value>> outerDimReplacementResult =
+      cloneAndReplaceDimInBackwardSlice(
+          rewriter, loc, dominanceInfo, outerUb, *this, expectedSparseDims[0],
+          [&](RewriterBase &rewriter, Location loc) {
+            OpFoldResult numRaggedRows = getNumRaggedRowsAsOfr();
+            Value numRaggedRowsVal =
+                getValueOrCreateConstantIndexOp(rewriter, loc, numRaggedRows);
+            return numRaggedRowsVal;
+          },
+          {outerUb});
+  if (failed(outerDimReplacementResult)) {
+    return emitOpError("failed to replace outer dim in backward slice");
+  }
+
+  auto outerFor = scf::ForOp::create(
+      rewriter, loc, outerLb, outerDimReplacementResult.value()[0], outerStep);
+  Value outerIv = outerFor.getInductionVar();
+  Block *outerForBody = outerFor.getBody();
+  rewriter.setInsertionPointToStart(outerForBody);
+
+  // For the inner sparse dimension, lower to loops by creating a loop with
+  // - Lower bound being `max(givenLowerBound, column_lengths[outerIv])`
+  // - Upper bound obtained by replacing the `memref.dim %sparseOp, 1` with
+  //   `column_lengthts[outerIv + 1]`
+  Value innerLb =
+      getValueOrCreateConstantIndexOp(rewriter, loc, givenRange[1].offset);
+  Value innerUb =
+      getValueOrCreateConstantIndexOp(rewriter, loc, givenRange[1].size);
+  Value innerStep =
+      getValueOrCreateConstantIndexOp(rewriter, loc, givenRange[1].stride);
+
+  // Replace the `memref.dim`/`tensor.dim` operation in the backward slice of
+  // the inner dim with the column length.
+  Value columnLengths = getColumnLengths();
+  FailureOr<SmallVector<Value>> innerDimReplacementResult =
+      cloneAndReplaceDimInBackwardSlice(
+          rewriter, loc, dominanceInfo, innerUb, *this, expectedSparseDims[1],
+          [&](RewriterBase &rewriter, Location loc) {
+            Value one = arith::ConstantIndexOp::create(rewriter, loc, 1);
+            Value plusOne = arith::AddIOp::create(rewriter, loc, outerIv, one);
+            Value columnUb =
+                memref::LoadOp::create(rewriter, loc, columnLengths, plusOne);
+            if (columnUb.getType() != innerUb.getType()) {
+              columnUb = arith::IndexCastOp::create(
+                  rewriter, loc, innerUb.getType(), columnUb);
+            }
+            return columnUb;
+          },
+          {innerLb, innerUb});
+  if (failed(innerDimReplacementResult)) {
+    return emitOpError("failed to replace inner dims in backward slice");
+  }
+
+  Value clonedLb = innerDimReplacementResult.value()[0];
+  Value clonedUb = innerDimReplacementResult.value()[1];
+  AffineExpr s0, s1;
+  bindSymbols(rewriter.getContext(), s0, s1);
+  AffineMap minMap = AffineMap::get(0, 2, {s0, s1}, rewriter.getContext());
```

Comment:
?
```suggestion
  AffineMap maxMap = AffineMap::get(0, 2, {s0, s1}, rewriter.getContext());
```
It is used in an affine.max below so the name looks off.